profile-widget/diveeventitem.cpp

```diff
@@ -254,20 +255,34 @@ bool DiveEventItem::shouldBeHidden()
 	return false;
 }
 
+int DiveEventItem::depthAtTime(int time)
+{
+	QModelIndexList result = dataModel->match(dataModel->index(0, DivePlotDataModel::TIME), Qt::DisplayRole, time);
+	if (result.isEmpty()) {
+		Q_ASSERT("can't find a spot in the dataModel");
+		hide();
+		return -666;
+	}
+	return dataModel->data(dataModel->index(result.first().row(), DivePlotDataModel::DEPTH)).toInt();
+}
+
 void DiveEventItem::recalculatePos(bool instant)
 {
 	if (!vAxis || !hAxis || !internalEvent || !dataModel)
 		return;
 
+
 	QModelIndexList result = dataModel->match(dataModel->index(0, DivePlotDataModel::TIME), Qt::DisplayRole, internalEvent->time.seconds);
 	if (result.isEmpty()) {
 		Q_ASSERT("can't find a spot in the dataModel");
 		hide();
 		return;
 	}
+	int depth = depthAtTime(internalEvent->time.seconds);
+	if (depth < 0)
```

Comment:
and then here you could compare with `==` to your symbolic constant

Reply:
I should definitely test against the constant (which should be negative enough not to be a possible value of a depth sensor)